### PR TITLE
update traffic-templates chart for kserve

### LINKED ISSUE #1438: Templatize Traffic Engineering solutions
The Iter8 traffic controller uses a `routemap` to determine what resources to watch and to guide the traffic configuration (for example, to redefine an Istio `VirtualService`). For common scenarios, the `routemap` might be templatized so that it is easier to generate/use.

Identify a strategy for using templates for the following scenarios:
- [x] blue-green (%-traffic split)
- [x] canary (request routing)

Assume use of Istio. It should work for KServe and for modelmesh.

Documentation of the traffic engineering features of Iter8 (https://github.com/iter8-tools/iter8/issues/1437) should use these templates.

### LINKED ISSUE #1520: Return version numbers as part of responses
**Is your feature request related to a problem? Please describe.**
Clients often need to know the version of the app from which they're getting a response.

**Describe the solution you'd like**
Modify the traffic templates (specifically, the VSes) to return the correct version number as part of response header.

### PR DESCRIPTION
**Pull request by kalantar**
Fixes [Return version numbers as part of response](https://github.com/iter8-tools/iter8/issues/1520)
Fixes [Templatize Traffic Engineering solutions](https://github.com/iter8-tools/iter8/issues/1438)

Required for [Document blue-green scenario for kserve serving](https://github.com/iter8-tools/iter8/issues/1509)
Required for [Document canary scenario for kserve serving](https://github.com/iter8-tools/iter8/issues/1510)

### CODE REVIEW

File: charts/traffic-templates/templates/traffic.yaml [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
-  {{- else if eq "canary" .Values.trafficStrategy }}
-    {{ include "routemap-canary" . }}
-  {{- else if eq "mirror" .Values.trafficStrategy }}
-    {{ include "routemap-mirror" . }}
-  {{- else }}
-    {{ fail "Unknown trafficStrategy"}}
+  {{- if eq "kserve-modelmesh" .Values.targetEnv }}
+    {{- if eq "blue-green" .Values.trafficStrategy }}
+      {{ include "routemap-bluegreen" . }}
+    {{- else {{- /* eq "canary" .Values.trafficStrategy */}}
```
Comment (Alan-Cha) [inline charts/traffic-templates/templates/traffic.yaml]: Is this else missing a closing bracket?

File: charts/traffic-templates/templates/traffic.yaml [hunk 2]
```diff
@@ -1,20 +1,56 @@
 {{- $versions := include "resolve.modelVersions" . | mustFromJson }}
 
+{{- /* validate values for templateName */}}
+{{- if not (has .Values.templateName (list "initialize-rollout" "modify-weights")) }}
+  {{- printf "Unknown templateName: '%s'" .Values.templateName | fail }}
+{{- end}}
```
Comment (Alan-Cha) [inline charts/traffic-templates/templates/traffic.yaml]:
```suggestion
{{- end }}
```

I see a few other instances.

File: charts/traffic-templates/templates/traffic.yaml (hunk 1 above)
Comment (kalantar) [reply charts/traffic-templates/templates/traffic.yaml]: fixed

File: charts/traffic-templates/templates/traffic.yaml (hunk 2 above)
Comment (kalantar) [reply charts/traffic-templates/templates/traffic.yaml]: fixed

**Review by Alan-Cha — APPROVED** [review]

**Pull request merged**